pkg/apis/options/login_url_parameters.go

```diff
@@ -0,0 +1,81 @@
+package options
+
+// LoginURLParameter is the configuration for a single query parameter that
+// can be passed through from the `/oauth2/start` endpoint to the IdP login
+// URL.  The "default" option specifies the default value or values (if any)
+// that will be passed to the IdP for this parameter, and "allow" is a list
+// of options for ways in which this parameter can be set or overridden via
+// the query string to `/oauth2/start`.
+// If _only_ a default is specified and no "allow" then the parameter is
+// effectively fixed - the default value will always be used and anything
+// passed to the start URL will be ignored.  If _only_ "allow" is specified
+// but no default then the parameter will only be passed on to the IdP if
+// the caller provides it, and no value will be sent otherwise.
+//
+// Examples:
+//
+// - A parameter whose value is fixed
+// ```
+// name: organization
+// default:
+// - myorg
+// ```
+//
+// - A parameter that is not passed by default, but may be set to one of a
+//   fixed set of values
+// ```
+// name: prompt
+// allow:
+// - value: login
+// - value: consent
+// - value: select_account
+// ```
+//
+// - A parameter that is passed by default but may be overridden by one of
+//   a fixed set of values
+// ```
+// name: prompt
+// default: ["login"]
+// allow:
+// - value: consent
+// - value: select_account
+// ```
+//
+// - A parameter that may be set to any email address
+// ```
+// name: login_hint
+// allow:
+// # Must contain an @ sign and at least one dot in the domain part
```

Comment:
Hmm, maybe I'll find out later, but, why do we need this restriction?

Reply:
I was just trying to think of an example that was a halfway house between "list of fixed strings" and "allow anything", but I guess yes, if users might read the example as implying that `login_hint` must _always_ be an email address then maybe a different example would be better.  I'll change it to something like "e.g. if you want to restrict `login_hint` to users in your own domain" and make the regex `^[^@]*@example\.com$` to be clearer that it's just an example rather than a spec requirement.